### ISSUE TITLE
Direction of the gradient

### ISSUE THREAD

**Issue by BBBBBBob** [alisiahkoohi/Langevin-dynamics#4]
Thanks for the code! I have a question regarding the gradient's direction when sampling the next variable.

In `class pSGLD(Optimizer)`, the mean is updated using a negative learning rate `p.data.addcdiv_(grad, G, value=-group['lr'])`. However, when calculating the transition probability in `def proposal_dist(self, idx: int) -> torch.Tensor` from `class MetropolisAdjustedLangevin(object)`, I think the gradient direction needs to be reversed `self.x[idx] - self.x[idx ^ 1] + self.lr_fn(self.counter) * self.grad[idx ^ 1]` instead of`self.x[idx] - self.x[idx ^ 1] - self.lr_fn(self.counter) * self.grad[idx ^ 1]`. Why is the gradient implemented in another direction?

**Comment by alisiahkoohi** [comment]
Thank you for your comments. To be consistent with the rest of the optimizers in `torch`, we consider the objective to be the negative log density. As a result, since we need to perform random walks around the modes of the target distribution, this translates to random walks around the valleys of the negative log density. Note that `pSGLD` does not use a negative learning rate; it uses a positive one but takes a gradient step towards the valley of the negative log density (the modes of the distribution). For reference, see the `RMSProp` implementation, which serves as the basis for `pSGLD`: https://github.com/pytorch/pytorch/blob/main/torch/optim/rmsprop.py#L326

**Comment by BBBBBBob** [comment]
> Thank you for your comments. To be consistent with the rest of the optimizers in `torch`, we consider the objective to be the negative log density. As a result, since we need to perform random walks around the modes of the target distribution, this translates to random walks around the valleys of the negative log density. Note that `pSGLD` does not use a negative learning rate; it uses a positive one but takes a gradient step towards the valley of the negative log density (the modes of the distribution). For reference, see the `RMSProp` implementation, which serves as the basis for `pSGLD`: https://github.com/pytorch/pytorch/blob/main/torch/optim/rmsprop.py#L326

Thank you for your clarification!
